lib/core/search-text/search-text-input.component.ts

```diff
@@ -279,4 +299,19 @@ export class SearchTextInputComponent implements OnInit, OnDestroy {
         this.onDestroy$.next(true);
         this.onDestroy$.complete();
     }
+
+    canShowClearSearch(): boolean {
+        return this.showClearButton && this.isSearchBarActive();
+    }
+
+    onClearSearch() {
+        if (this.defaultState === SearchTextStateEnum.collapsed) {
```

Comment:
I think it would be better to have a method 
isSearchStateCollapsed
isSearchStateExpanded